vpr/src/base/read_route.cpp

```diff
@@ -411,6 +413,35 @@ static void format_coordinates(int& x, int& y, std::string coord, ClusterNetId n
     }
 }
 
+// ptc (Pin-Track-Class) can be one of below patterns in route graph
+//  Node:   714 SOURCE (4,1)  Class: 18  Switch: 0
+//  Node:   728   OPIN (4,1)  Pin: 57   clb.O[17] Switch: 2
+//  Node:   16733    CHANX (6,2) to (9,2)  Track: (56,58,60,62)  Switch: 1
+//  Node:   23176    CHANY (4,1) to (4,3)  Track: (69,67,65)  Switch: 1
+static void format_ptc_num(int& n0, int& n1, int& n2, int& n3, std::string ptc_str, ClusterNetId net, const char* filename, const int lineno) {
+    // detect how many number needs to be extracted
+    size_t ptc_count = std::count(ptc_str.begin(), ptc_str.end(), ',') + 1;
+    // detect and remove the parenthesis
+    std::stringstream ptc_stream(ptc_count > 1? format_name(ptc_str) : ptc_str);
+    for (size_t i = 0; i < ptc_count; i++) {
```

Comment:
May consider to use the tokenizer class: 

https://github.com/lnis-uofu/OpenFPGA/blob/master/libopenfpga/libopenfpgautil/src/openfpga_tokenizer.h